velox/connectors/hive/benchmarks/HivePartitionFunctionBenchmark.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/connectors/hive/HivePartitionFunction.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+namespace facebook::velox::test {
+namespace {
+
+void BM_HivePartFunc(
+    size_t,
+    size_t vectorSize,
+    size_t bucketCount,
+    const TypePtr& type,
+    size_t stringLength = 0) {
+  functions::test::FunctionBenchmarkBase benchmarkBase;
+  folly::BenchmarkSuspender suspender;
+
+  VectorFuzzer::Options opts;
+  opts.vectorSize = vectorSize;
+  opts.stringLength = stringLength;
+  VectorFuzzer fuzzer(opts, benchmarkBase.pool());
+  auto vector = fuzzer.fuzzRow(ROW({type}));
+
+  std::vector<int> bucketToPartition(bucketCount);
+  std::vector<ChannelIndex> keyChannels;
+  std::vector<uint32_t> partitions(vectorSize);
+
+  std::iota(bucketToPartition.begin(), bucketToPartition.end(), 0);
+  keyChannels.emplace_back(0);
+  connector::hive::HivePartitionFunction partitionFunction(
+      bucketCount, bucketToPartition, keyChannels);
+
+  suspender.dismiss();
+
+  partitionFunction.partition(
+      *reinterpret_cast<RowVector*>(vector.get()), partitions);
+}
+
+} // namespace
+
+BENCHMARK_NAMED_PARAM(
+    BM_HivePartFunc,
+    BOOLEAN_10M_ROWS_1K_BUCKETS,
+    10'000'000,
```

Comment:
10M rows and 1K partition parameter values are always the same. In practice, we do not see such large vectors. Hence, better, benchmark 1K rows and 10K rows with 20 and 100 partitions.

Reply:
Ideally, we'd test 4 combinations [1K, 10K] x [20, 100] since the vector sizes and partition counts are independent of each other.

Reply:
> I tested 20/100 partition against the same row number and there is no perf difference.

That's good and expected.

Reply:
Got it, thanks!